frontend/awx/access/common/useViewActivityStream.tsx

```diff
@@ -8,20 +8,59 @@ import {
   usePageNavigate,
 } from '../../../../framework';
 import { AwxRoute } from '../../main/AwxRoutes';
-
-export function useViewActivityStream<T extends object>(type: string) {
+import { useAwxConfig } from '../../common/useAwxConfig';
+import { useLocation } from 'react-router-dom';
+const topLevelPath = ['administration', 'analytics', 'infrastructure', 'access', 'settings'];
+export function useViewActivityStream<T extends object>() {
   const { t } = useTranslation();
   const pageNavigate = usePageNavigate();
-  return useMemo<IPageAction<T>[]>(
-    () => [
+  const config = useAwxConfig();
+  const location = useLocation();
+  const firstURLItems = location.pathname.split('/')[1];
+  const secondURLItems = location.pathname.split('/')[2];
+  const isSchedule = location.pathname.split('/').includes('schedules');
+
+  const pathIdentifier = isSchedule
+    ? 'schedules'
+    : topLevelPath.includes(firstURLItems)
+      ? secondURLItems
+      : firstURLItems;
+  let type: string;
+
+  switch (pathIdentifier) {
+    case 'inventories':
+      type = 'inventory';
+      break;
+    case 'schedules':
+      type = 'schedule';
+      break;
+    case 'applications':
+      type = 'o_auth2_application';
+      break;
+    case 'notifiers':
+      type = 'notification_template';
+      break;
+    case 'credential-types':
+      type = 'credential_type';
+      break;
+    case 'instance-groups':
+      type = 'instance_group';
+      break;
+    default:
+      type = pathIdentifier.slice(0, -1);
```

Comment:
What is the purpose of the slice here? Isn't `pathIdentifier` always going to be a string?